x-pack/plugins/canvas/canvas_plugin_src/functions/common/alterColumn.ts

```diff
@@ -96,8 +98,8 @@ export function alterColumn(): ExpressionFunctionDefinition<
       }
 
       const rows = input.rows.map((row) => ({
-        ...omit(row, column.name),
-        [name]: handler(row[column.name]),
+        ...omit(row, prevColumnId),
+        [id]: handler(row[prevColumnId]),
```

Comment:
I'm trying to think through if there could be any unintended side effects of not also changing the id? 

Only scenario I could think through is something like 

```
alterColumn "old-column" name="altered-column" 
| staticColumn "old-column" 
```

Is that now going to overwrite the id "old-column" where as previously you would end up with "altered-column" and "old-column"

Reply:
Yea, but this is raising the other question. Do we need to expose separately rewriting of `ID` and `name`.
Let us imagine that we are going to use some external library and we don't want to look through the docs. We want to take the function and pass there some params and get the expected result. 

From the name of the function, we can see, that it is updating the column. The `column` arg is a selector, that makes sense. And then we are passing the `name` argument. Just from the name of the argument, we can expect, that the expression function will update the `name` of the column, without updating the `ID`.

So, the question here is should we provide a separate mechanism for updating the `ID`. 

I just want to prevent increasing unexpected and confusing behavior.